### PR TITLE
add Serum.Plugins.SitemapGenerator

### PR DESCRIPTION
**Pull request by kernelgarden**
I needed to expose my blog to the Google search engine. 
I thought it would be nice to have a plug-in like [jekyll-sitemap](https://github.com/jekyll/jekyll-sitemap) of Jekyll in Serum" And I made this plug-in.

### CODE REVIEW

**Comment by Dalgona** [comment]
Awesome! I'll take a look at your code when I have some time.

Actually you could just publish your plugin to hex.pm, and I have a plan to make a page that features some third-party plugins. But, yes, I also think it's good to have this plugin in this repository. Thanks!

**Comment by kernelgarden** [comment]
I was able to make a blog because you made a good static website generator. I think I made [my blog](https://kernelgarden.github.io) interesting. Thank you!

File: lib/serum/plugins/sitemap_generator.ex [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    |> IO.iodata_to_binary()
+    |> Path.join("build_resources/#{resource}")
+    |> File.read!()
+  end
+
+  defp write_sitemap(dest) do
+    all_posts = GlobalBindings.get(:all_posts)
+
+    read_build_resource("sitemap_template.xml")
+    |> EEx.eval_string(
```
Comment (Dalgona) [inline lib/serum/plugins/sitemap_generator.ex]: You can embed precompiled templates into the module at compile time for runtime performance by using [`EEx.function_from_file/5`](https://hexdocs.pm/eex/EEx.html#function_from_file/5) or [`EEx.function_from_string/5`](https://hexdocs.pm/eex/EEx.html#function_from_string/5).

File: priv/build_resources/sitemap_template.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      <loc><%= Path.join(@server_root, post.url) %></loc>
+
+      <lastmod><%= @transformer.(post.raw_date) %></lastmod>
+      
+      <changefreq>weekly</changefreq>
+
+      <priority>1.0</priority>
+    </url>
+  <% end %>
+</urlset>
```
Comment (Dalgona) [inline priv/build_resources/sitemap_template.xml]: Please make sure that there is a newline character at the very end of the file.

File: lib/serum/plugins/sitemap_generator.ex [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      ]
+    )
+    |> try_save(Path.join(dest, "sitemap.xml"))
+  end
+
+  defp write_robots(dest) do
+    sitemap_path = Path.join(get_server_root(), "sitemap.xml")
+
+    read_build_resource("robots_template.txt")
+    |> EEx.eval_string(assigns: [sitemap_path: sitemap_path])
```
Comment (Dalgona) [inline lib/serum/plugins/sitemap_generator.ex]: Same thing as line 66 :)

**Comment by Dalgona** [comment]
Looks good to me. I'll do some refactoring after the merge. Please review my comments FYI.

Thank you again!!

**Pull request merged**